support/testing/scenarios/setup.go

```diff
@@ -0,0 +1,150 @@
+	initState.NextID = abi.ActorID(builtin.FirstNonSingletonActorId + additionalAccounts)
+	initializeActor(ctx, t, vm, initState, builtin.InitActorCodeID, builtin.InitActorAddr, big.Zero())
+
+	rewardState := reward.ConstructState(big.Mul(big.NewInt(100), FIL))
+	initializeActor(ctx, t, vm, rewardState, builtin.RewardActorCodeID, builtin.RewardActorAddr, big.Max(big.NewInt(14e8), FIL))
+
+	cronState := cron.ConstructState(cron.BuiltInEntries())
+	initializeActor(ctx, t, vm, cronState, builtin.CronActorCodeID, builtin.CronActorAddr, big.Zero())
+
+	powerState := power.ConstructState(emptyMapCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, powerState, builtin.StoragePowerActorCodeID, builtin.StoragePowerActorAddr, big.Zero())
+
+	marketState := market.ConstructState(emptyMapCID, emptyArrayCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, marketState, builtin.StorageMarketActorCodeID, builtin.StorageMarketActorAddr, big.Zero())
+
+	vrState := verifreg.ConstructState(emptyMapCID, rootVerifier.idAddr)
+	initializeActor(ctx, t, vm, vrState, builtin.VerifiedRegistryActorCodeID, builtin.VerifiedRegistryActorAddr, big.Zero())
+
+	// burnt funds
+	initializeActor(ctx, t, vm, &account.State{Address: builtin.BurntFundsActorAddr}, builtin.AccountActorCodeID, builtin.BurntFundsActorAddr, big.Zero())
+
+	// verified root
+	rootVerifierState := &account.State{Address: rootVerifier.pubAddr}
+	initializeActor(ctx, t, vm, rootVerifierState, builtin.AccountActorCodeID, rootVerifier.idAddr, big.Zero())
+
+	// additional accounts
+	initialAccountBalance := big.Mul(big.NewInt(10_000), FIL)
+	pubAddrs := make([]address.Address, len(accountAddrs))
+	for i, addrPair := range accountAddrs {
+		st := &account.State{Address: addrPair.pubAddr}
+		initializeActor(ctx, t, vm, st, builtin.AccountActorCodeID, addrPair.idAddr, initialAccountBalance)
+		pubAddrs[i] = addrPair.pubAddr
+	}
+
+	_, err = vm.commit()
+	require.NoError(t, err)
+
+	return vm, pubAddrs
+}
+
+func initializeActor(ctx context.Context, t *testing.T, vm *VM, state runtime.CBORMarshaler, code cid.Cid, a address.Address, balance abi.TokenAmount) {
+	stateCID, err := vm.store.Put(ctx, state)
+	require.NoError(t, err)
+	actor := &TestActor{
+		Head:    stateCID,
+		Code:    code,
+		Balance: balance,
+	}
+	err = vm.setActor(ctx, a, actor)
+	require.NoError(t, err)
+}
+
+type addrPair struct {
+	pubAddr address.Address
+	idAddr  address.Address
+}
+
+func initAddressTree(t *testing.T, vm *VM, rootVerifier addrPair, additionalAccounts int) (cid.Cid, []addrPair) {
+	m := adt.MakeEmptyMap(vm.store)
+	err := m.Put(adt.AddrKey(rootVerifier.pubAddr), cborAddrID(t, rootVerifier.idAddr))
+	require.NoError(t, err)
+
+	newAccounts := make([]addrPair, additionalAccounts)
+	for i := 0; i < additionalAccounts; i++ {
+		addr := actor_testing.NewBLSAddr(t, 93837778)
+		require.NoError(t, err)
+		idAddr, err := address.NewIDAddress(uint64(builtin.FirstNonSingletonActorId + i))
```

Comment:
It would be much better if this just used `init.State.MapAddressToNewID`. This seems to be reaching around the existing code unnecessarily.